pkg/app/ui/handler.go

```diff
@@ -0,0 +1,106 @@
+package ui
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/aserto-dev/topaz/pkg/cc/config"
+)
+
+type fsWithDefinition struct {
+	consoleFS http.FileSystem
+}
+
+func (f *fsWithDefinition) Open(name string) (http.File, error) {
+	if strings.HasPrefix(name, "/ui/") {
+		return f.consoleFS.Open("console/build/index.html")
+	}
+
+	name = strings.TrimPrefix(name, "/public")
+	return f.consoleFS.Open(fmt.Sprintf("console/build%s", name))
+}
+
+func UIHandler(consoleFS http.FileSystem) http.Handler {
+	return http.FileServer(&fsWithDefinition{consoleFS: consoleFS})
+}
+
+func ConfigHandler(confServices *config.Config) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		type consoleCfg struct {
+			AsertoDirectoryURL   string `json:"asertoDirectoryUrl"`
+			AuthorizerServiceURL string `json:"authorizerServiceUrl"`
+			AuthorizerAPIKey     string `json:"authorizerApiKey"`
+			DirectoryAPIKey      string `json:"directoryApiKey"`
+			DirectoryTenantID    string `json:"directoryTenantId"`
+		}
+
+		var apiKey string
+		for key := range confServices.Auth.APIKeys {
+			apiKey = key
+			break
+		}
+
+		cfg := &consoleCfg{}
+		if serviceConfig, ok := confServices.Services["authorizer"]; ok {
+			cfg.AuthorizerServiceURL = fmt.Sprintf("https://%s", serviceConfig.Gateway.ListenAddress)
+			cfg.AuthorizerAPIKey = apiKey
+		}
+
+		if serviceConfig, ok := confServices.Services["reader"]; ok {
+			cfg.AsertoDirectoryURL = fmt.Sprintf("https://%s", serviceConfig.Gateway.ListenAddress)
+		} else {
+			host := strings.Split(confServices.DirectoryResolver.Address, ":")[0]
```

Comment:
I guess the console needs to be able to accept different addresses for the reader and writer because users can configure them to run on different ports in topaz.